### DISCUSSION TITLE
Cypress Screenshot naming logic

### DISCUSSION THREAD

**Discussion by cbruge** [cypress-io/cypress#16005] [Questions and Help] [answered]
Hey all,

can someone point me in the right direction of where the screenshot naming logic of Cypress is defined?
I am trying to addContext to the mochawesome report with the test:after:run event hook.
It works fine for basic cases but, e.g. context within context, beforeAll hook failures etc...they are all considered in the screenshot naming.
Renaming the screenshot to something also does not work with the after:screenshot because if several tests in the same context fail the only differentiating factor is the timestamp which the test:after:run hook has no access to.

Current logic, which covers most but not all cases:

Cypress.on("test:after:run", (test, runnable) => {
  const runnableParentTitle = runnable.parent.title.replace("/", "");
  const testTitle = test.title.replace("/", "");
  let screenshot = `./screenshots/${Cypress.spec.name}/${runnableParentTitle} -- ${testTitle} (failed).png`;
  if (!runnable.parent.title) {
    screenshot = `./screenshots/${Cypress.spec.name}/${testTitle} (failed).png`;
  }
  if (test.state === "failed") {
    addContext({ test }, screenshot);
  }
});

Any ideas?😅

**Comment by cbruge** [comment]
Just found out the runnable.parent can also have a parent...and from this i can assert the other naming...but is there a more elegant way, e.g something with fullTitle?

**Comment by bahmutov** [comment, marked as answer]
The code responsible for saving the screenshots is in https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/screenshots.js and our naming scheme is described in https://on.cypress.io/screenshot#Naming-conventions

**Reply by cbruge** [reply]
Thank you for pointing me in the right direction!
